src/collection_manager.cpp

```diff
@@ -528,9 +528,36 @@ Option<nlohmann::json> CollectionManager::drop_collection(const std::string& col
 
     s_lock.unlock();
 
+    auto embedding_fields = collection->get_embedding_fields();
+
     std::unique_lock u_lock(mutex);
     collections.erase(actual_coll_name);
     collection_id_names.erase(collection->get_collection_id());
+
+    for(auto& embedding_field : embedding_fields) {
```

Comment:
This logic looks be duplicated in `remove_embedding_field`. Should consolidate into a function.